### PR TITLE
Add Styx API Overview page to the developer docs.

### PR DESCRIPTION
**Pull request by mikkokar**

### CODE REVIEW

File: docs/developer-guide/api-overview.md [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+        @Override
+        protected HttpResponse doHandle(HttpRequest request) {
+            return PONG.toStreamingResponse();
+        }
+    }   
+```
+ 
+### Http Interceptor Interface
+
+A HTTP interceptor is an object transforms, responds, or runs side-effecting actions
```
Comment (dvlato) [inline docs/developer-guide/api-overview.md]: object **that**

File: docs/developer-guide/api-overview.md [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+        return chain.proceed(newRequest)
+                .map(response -> response.newBuilder()
+                        .header(VIA, viaHeader(response))
+                        .build());
+    }
+```
+
+The chain also contains a request context which can be obtained with a 
+call to `chain.context()`. It is a set of key-value properties associated 
+with the request. Plugins may store some information in the context. So does
```
Comment (dvlato) [inline docs/developer-guide/api-overview.md]: Review the  "So does styx core..." bit?

File: docs/developer-guide/api-overview.md [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+thread will block the Styx event loop.  
+
+
+### Styx Observable
+
+Conceptually similar to Futures, `StyxObservable` is a data type that 
+facilitates asynchronous event handling, modelled after Rx 
+[observables](http://reactivex.io/documentation/observable.html).
+
+However `rx.Observable` is very generic reactive stream abstraction.
```
Comment (dvlato) [inline docs/developer-guide/api-overview.md]: is **a** very

File: docs/developer-guide/api-overview.md [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+few lines of code. The `styx-client` can be used as a reactive HTTP client.
+
+The `styx-api-testsupport` provides helpers for unit testing Styx extensions.
+ 
+
+The relevant modules are available in Maven Central:
+
+  * Styx API: https://mvnrepository.com/artifact/com.hotels.styx/styx-api
+  * Styx API Test Support: https://mvnrepository.com/artifact/com.hotels.styx/styx-api-testsupport
+  * Styx Server: https://mvnrepository.com/artifact/com.hotels.styx/styx-server
```
[the remark this answers is not in the document]
Comment (mikkokar) [reply docs/developer-guide/api-overview.md]: Yes. Lets do that as a separate increment.

File: docs/developer-guide/api-overview.md [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+Therefore a `HttpRequest` (`HttpResponse`) can be aggreaged to a full 
+message strictly once only.
+
+The type signature illustrates other facts about the content stream:
+
+* Return type of `StyxObservable` shows that HTTP content stream is 
+  aggregated asynchronously. Obviously because the rest of the content 
+  stream are yet to be received from the network.
+
+* The `maxContentBytes` is the maximum sets the upper limit for the message
```
[the remark this answers is not in the document]
Comment (mikkokar) [reply docs/developer-guide/api-overview.md]: sure

File: docs/developer-guide/api-overview.md [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+corresponding `FullHttpRequest` (`FullHttpResponse`). Its type signature is:
+
+```java
+   public StyxObservable<FullHttpRequest> toFullRequest(int maxContentBytes);
+```
+
+Note that the HTTP content streams represents live network traffic streaming
+through Styx core. The content stream is not memoized nor stored. 
+When the content is gone, it is gone for good. 
+Therefore a `HttpRequest` (`HttpResponse`) can be aggreaged to a full
```
[the remark this answers is not in the document]
Comment (mikkokar) [reply docs/developer-guide/api-overview.md]: ok

File: docs/developer-guide/api-overview.md [hunk 7]
```diff
@@ -0,0 +1,229 @@
...
+  bodies, or perhaps from denial of service attacks. If the limit is reached,
+  the conversion fails with `ContentOverflowException`. 
+
+Converting a full message to a stream is much easier. The `FullHttpRequest` 
+(`FullHttpResponse`) has a method called `toStreamingRequest` (or `toStreamingResponse`): 
+```java
+    public HttpRequest toStreamingRequest()
+``` 
+
+As the method sinature reveals a `HttpResponse` is available immediately.
```
[the remark this answers is not in the document]
Comment (mikkokar) [reply docs/developer-guide/api-overview.md]: 👍

File: docs/developer-guide/api-overview.md (hunk 1 above)
Comment (mikkokar) [reply docs/developer-guide/api-overview.md]: 👍

**Pull request merged**